### PR TITLE
User collections

### PR DESCRIPTION
**Pull request by behica**
Added a collection page and the ability for users to favorite / add stories to their collection.

### CODE REVIEW

File: app/controllers/collections_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		
+		if collection.destroy
+       flash[:notice] = "Story removed from collection."
+    else
+       flash[:alert] = "Something went wrong. Try again."
+		end
+    
+    redirect_to :back
+	end
+end
```
Comment (bobbilee19) [inline app/controllers/collections_controller.rb]: why is the spacing so messed up ?

File: app/models/story.rb [hunk 2]
```diff
@@ -1,2 +1,3 @@
 class Story < ActiveRecord::Base
+	has_many :collections, dependent: :destroy
 end
```
Comment (bobbilee19) [inline app/models/story.rb]: spacing too far indented here too

File: app/models/user.rb [hunk 3]
```diff
@@ -3,4 +3,11 @@ class User < ActiveRecord::Base
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable, :confirmable
+         
+  has_many :stories
+  has_many :collections, dependent: :destroy
+  
+  def collection_for(story)
+    collections.where(story_id: story.id).first
```
Comment (bobbilee19) [inline app/models/user.rb]: this is kind of a weird name for a method when i read it outloud.

and not a good idea to do .first on something

what are u thinking with this one? what's it for?

File: app/views/collections/_collect.html.erb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<% if collect = current_user.collection_for(story) %>
```
Comment (bobbilee19) [inline app/views/collections/_collect.html.erb]: if this is something that they collected?

I wouldn't do these kind of condtionals inside view it should be in a controller.

File: app/views/collections/_collect.html.erb [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+	<%= link_to [story, collect], class: 'btn btn-danger btn-sm', method: :delete do %>
+		<span class="glyphicon glyphicon-heart"></span>
+	<% end %>
+<% else %>
+	<%= link_to [story, Collection.new], class: 'btn btn-default btn-sm', method: :story do %>
+		<span class="glyphicon glyphicon-heart-empty"></span>
+	<% end %>
+<% end %>
```
Comment (bobbilee19) [inline app/views/collections/_collect.html.erb]: spacing is indented too far in this whole file too

File: app/views/collections/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<h1>My Collection</h1>
+<% if current_user.collections.empty? %>
```
Comment (bobbilee19) [inline app/views/collections/index.html.erb]: this should be an easy @collections.empty? - do the other stuff in controller

File: app/controllers/collections_controller.rb (hunk 1 above)
Comment (behica) [inline app/controllers/collections_controller.rb]: So odd, looks fine on c9.io. Seems this happens a bit when I push files from there.

File: app/models/user.rb (hunk 3 above)
Comment (behica) [inline app/models/user.rb]: This is based on the Bloccit `favorite_for` method. Basically it creates an array if the user has added a story (based on `story_id`) to their collection, and shows as nil if they have not. Should I use a different method?

File: app/views/collections/_collect.html.erb (hunk 4 above)
Comment (behica) [inline app/views/collections/_collect.html.erb]: OK, I'll look into moving it to controller. The partial is called within the collection index view, where it searches the current user's collections and checks if they're empty or not. If empty, it displays a message stating there is nothing in collection. If it has something, it goes through the `current_user.collections.each do |collection|` logic and applies the icon change button.

File: app/views/collections/_collect.html.erb (hunk 5 above)
Comment (behica) [inline app/views/collections/_collect.html.erb]: It's at 2 spaces on the file, hopefully next push clears it.

File: app/views/collections/index.html.erb (hunk 6 above)
Comment (behica) [inline app/views/collections/index.html.erb]: OK, I'll define that in the controller.

**Pull request merged**